airbyte-cron/src/main/java/io/airbyte/cron/config/DatabaseBeanFactory.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cron.config;
+
+import io.airbyte.config.persistence.ConfigPersistence;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.DatabaseConfigPersistence;
+import io.airbyte.config.persistence.split_secrets.JsonSecretsProcessor;
+import io.airbyte.db.Database;
+import io.airbyte.db.check.DatabaseMigrationCheck;
+import io.airbyte.db.factory.DatabaseCheckFactory;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Value;
+import io.micronaut.flyway.FlywayConfigurationProperties;
+import java.io.IOException;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.sql.DataSource;
+import lombok.extern.slf4j.Slf4j;
+import org.flywaydb.core.Flyway;
+import org.jooq.DSLContext;
+
+/**
+ * Micronaut bean factory for database-related singletons.
+ */
+@Factory
+@Slf4j
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public class DatabaseBeanFactory {
+
+  private static final String BASELINE_DESCRIPTION = "Baseline from file-based migration v1";
+  private static final Boolean BASELINE_ON_MIGRATION = true;
+  private static final String INSTALLED_BY = "AirbyteCron";
+
+  @Singleton
+  @Named("configDatabase")
+  public Database configDatabase(@Named("config") final DSLContext dslContext) throws IOException {
```

Comment:
@jdpgrailsdev Won't this be conflicting with the Bean declaration that is added in your review? It will complains that multiple Beans with this type are present.

Reply:
@benmoriceau I'm not sure how Micronaut handles that.  It may just replace it.  We are going to have to work out the duplication once everything lands, most likely by extracting it to a shared location.

Reply:
Just to be clear here: is the recommendation then that it's ok to merge as-is and de-dupe when both are merged? If so I would love a general review / approval from either of you on the platform team 😄